cognite/client/_api/containers.py

```diff
@@ -0,0 +1,179 @@
+
+    def list(
+        self, space: str | None = None, limit: int = CONTAINER_LIST_LIMIT_DEFAULT, include_global: bool = False
+    ) -> ContainerList:
+        """`List containers <https://docs.cognite.com/api/v1/#tag/Containers/operation/listContainers>`_
+
+        Args:
+            space (int, optional): The space to query
+            limit (int, optional): Maximum number of containers to return. Default to 10. Set to -1, float("inf") or None
+                to return all items.
+            include_global (bool, optional): Whether the global containers should be returned.
+
+        Returns:
+            ContainerList: List of requested containers
+
+        Examples:
+
+            List containers and limit to 5:
+
+                >>> from cognite.client import CogniteClient
+                >>> c = CogniteClient()
+                >>> container_list = c.data_modeling.containers.list(limit=5)
+
+            Iterate over containers::
+
+                >>> from cognite.client import CogniteClient
+                >>> c = CogniteClient()
+                >>> for container in c.data_modeling.containers:
+                ...     container # do something with the container
+
+            Iterate over chunks of containers to reduce memory load::
+
+                >>> from cognite.client import CogniteClient
+                >>> c = CogniteClient()
+                >>> for container_list in c.data_modeling.containers(chunk_size=10):
+                ...     container_list # do something with the containers
+        """
+        filter = ContainerFilter(space, include_global)
+
+        return self._list(
+            list_cls=ContainerList,
+            resource_cls=Container,
+            method="GET",
+            limit=limit,
+            filter=filter.dump(camel_case=True),
+        )
+
+    @overload
+    def apply(self, container: Sequence[Container]) -> ContainerList:
+        ...
+
+    @overload
+    def apply(self, container: Container) -> Container:
+        ...
+
+    def apply(self, container: Container | Sequence[Container]) -> Container | ContainerList:
+        """`Add or update (upsert) containers. <https://docs.cognite.com/api/v1/#tag/Containers/operation/ApplyContainers>`_
+
+        Args:
+            container (container: Container | Sequence[Container]): Container or containers of containers to create or update.
+
+        Returns:
+            Container | ContainerList: Created container(s)
+
+        Examples:
+
+            Create new containersd:
+
+                >>> from cognite.client import CogniteClient
+                >>> import cognite.client.data_classes.data_modeling as models
+                >>> c = CogniteClient()
+                >>> containers = [models.Container(space="mySpace", container="myContainer",
+                ... properties={"name": models.ContainerPropertyIdentifier(type=models.TextType, name="name")})]
```

Comment:
Doesn't the use of ellipsis make this fail in our doctests?

If not, then I think you need to add the new APIs to doctests so that they are run.

Reply:
Isn't this correct syntax? Ellipsis used for multiline in doc comments, [ref](https://stackoverflow.com/questions/30448746/multiline-statements-using-python-doctest).

Reply:
No idea 😂 Ty for link

Reply:
Oh never mind my comment; complete brain fart 😂